### PR TITLE
Fixed exception with YamlGenerator when use list with placeholders

### PR DESCRIPTION
**Pull request by altro3**
Now you can't set placeholders in list property like this:
```yaml
otel:
  exclusions:
    - ${my.prop1}
    - ${my.prop2}
```
You will see this exception:
```
Caused by: java.lang.IllegalArgumentException: index 1 for '${' not in range (received 0 arguments)
	at com.squareup.javapoet.Util.checkArgument(Util.java:53)
	at com.squareup.javapoet.CodeBlock$Builder.add(CodeBlock.java:287)
	at com.squareup.javapoet.CodeBlock$Builder.addStatement(CodeBlock.java:402)
	at com.squareup.javapoet.MethodSpec$Builder.addStatement(MethodSpec.java:525)
	at io.micronaut.aot.core.codegen.MapGenerator.generateListMethod(MapGenerator.java:100)
	at io.micronaut.aot.core.codegen.MapGenerator.convertValueToSource(MapGenerator.java:61)
	at io.micronaut.aot.core.codegen.MapGenerator.generateMap(MapGenerator.java:44)
	at io.micronaut.aot.std.sourcegen.MapPropertySourceGenerator.generate(MapPropertySourceGenerator.java:77)
```

This PR fixes bug.

Also fixed project to run it on Windows

### CODE REVIEW

**Comment by altro3** [comment]
@melix Hi! Please, review this fix. It's critical bug, I think

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy [hunk 1]
```diff
@@ -66,29 +66,31 @@ abstract class AbstractSourceGeneratorSpec extends Specification {
...
         def sourceGenerator = newGenerator()
         sourceGenerator.generate(context)
-        def sources = context.getGeneratedJavaFiles().collectEntries([:]) {
+        def javaFiles = context.getGeneratedJavaFiles()
+        def sources = new HashMap<JavaFile, String>()
+        javaFiles.forEach {
             def writer = new StringWriter()
             it.writeTo(writer)
-            [it, writer.toString()]
+            sources.put(it, writer.toString())
```
Comment (melix) [inline aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy:74]: question: Why did you change this?

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy [hunk 2]
```diff
@@ -66,29 +66,31 @@ abstract class AbstractSourceGeneratorSpec extends Specification {
...
         }
 
         this.generatedSources = new GeneratedSources(sources, context.getGeneratedStaticInitializers(), resourcesDir.toFile())
     }
 
     void assertThatGeneratedSources(@DelegatesTo(value=GeneratedSources, strategy = Closure.DELEGATE_FIRST) Closure<?> spec) {
         if (generatedSources == null) {
-            throw new AssertionError("Couldn't find generated sources. Did you call 'generateSources()'?")
+            throw new AssertionError((Object) "Couldn't find generated sources. Did you call 'generateSources()'?")
```
Comment (melix) [inline aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy:82]: suggestion: I would prefer you revert those changes. There is no problem with the original code, I can only think you're doing this because the IDE messes up, but it shouldn't.

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy [hunk 3]
```diff
@@ -66,29 +66,31 @@ abstract class AbstractSourceGeneratorSpec extends Specification {
...
         }
         spec.delegate = generatedSources
         spec.resolveStrategy = Closure.DELEGATE_FIRST
         spec()
         def verified = generatedSources.verifiedClasses
         def missing = generatedSources.generatedClasses - verified
         if (missing) {
-            throw new AssertionError("Too few assertions. You also need to verify classes: $missing")
+            throw new AssertionError((Object) "Too few assertions. You also need to verify classes: $missing")
```
Comment (melix) [inline aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy:90]: suggestion: remove those nasty casts to `Object`, to keep readability and to reduce the diff size of this fix

File: gradle.properties [hunk 4]
```diff
@@ -1,8 +1,9 @@
 projectVersion=1.1.2-SNAPSHOT
-micronautVersion=3.3.0
-micronautTestVersion=2.3.7
-groovyVersion=3.0.9
-spockVersion=2.0-groovy-3.0
+micronautVersion=3.8.5
```
Comment (melix) [inline gradle.properties:2]: issue: by upgrading the different libraries we can't make sure that test resources will remain compatible with 3.3+
suggestion: revert the change, test resources 2.0.0 will bump to Micronaut 4 as the baseline

**Review by melix — CHANGES_REQUESTED** [review]
Thanks a lot for the fix! LGTM overall but most of this PR is actually not about the fix itself. To make things easier to merge and review, I would recommend to keep the PR as small as possible with the minimal set of changes.

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy (hunk 2 above)
Comment (altro3) [reply aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy:82]: Hmm,, maybe it was a bug in IDEA, but it's highlighted these lines in red for me, because it was not possible to determine the correct constructor and she indicated that the constructor with the string is private. Okay, I'll remove these changes. 
изображение

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy (hunk 1 above)
Comment (altro3) [reply aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy:74]: The problem with invalid generics. The resulting object, for some reason `Map<Object, Object>`. Because of this, an error occurs when trying to compile, because it requires `Map<JavaFile, String>`

Maybe this is specific of jdk or groovy for windows

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy (hunk 3 above)
Comment (altro3) [reply aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy]: reverted

File: gradle.properties (hunk 4 above)
Comment (altro3) [reply gradle.properties]: done

File: settings.gradle.kts [hunk 5]
```diff
@@ -23,7 +23,9 @@ include("aot-api")
 include("aot-cli")
 
 val micronautVersion = providers.gradleProperty("micronautVersion")
-        .forUseAtConfigurationTime()
+    .forUseAtConfigurationTime()
```
[the remark this answers is not in the document]
Comment (altro3) [reply settings.gradle.kts]: done

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy (hunk 1 above)
Comment (melix) [reply aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy:74]: What do you use to compile? The Gradle build just passes fine, so this hints at an IDEA bug

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy (hunk 2 above)
Comment (melix) [reply aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy]: Yes it's a false positive in IDEA

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy (hunk 1 above)
Comment (altro3) [reply aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy:74]: No, this is not a problem of IDEA. Look to screenshots. First - run from IDEA, second - from powershell:
изображение
изображение

File: gradle.properties [hunk 6]
```diff
@@ -1,8 +1,8 @@
 projectVersion=1.1.2-SNAPSHOT
 micronautVersion=3.3.0
 micronautTestVersion=2.3.7
-groovyVersion=3.0.9
-spockVersion=2.0-groovy-3.0
+groovyVersion=3.0.15
```
Comment (melix) [inline gradle.properties:4]: nitpick: Can you revert those as well? I think we should really reduce to the minimal fix, to make things easier.

Comment (altro3) [reply gradle.properties]: done

File: aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy (hunk 1 above)
Comment (altro3) [reply aot-core/src/testFixtures/groovy/io/micronaut/aot/core/codegen/AbstractSourceGeneratorSpec.groovy]: @melix hah, really! Reverted groovy version and now all works fine. Reverted these changes

**Review by melix — APPROVED** [review]
